mne/io/nedf/nedf.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+"""Import NeuroElectrics DataFormat (NEDF) files"""
+
+import datetime
+import os
+from xml.etree import ElementTree
+
+import numpy as np
+
+from .. import BaseRaw
+from ... import create_info, Annotations
+
+
+def parse_nedf_header(filename):
+    """
+    Read the header information from the first 10kB of an .nedf file
+
+    Parameters
+    ----------
+    filename : str
+        Path to the .nedf file
+
+    Returns
+    -------
+    info : dict
+        A dictionary with information from the header
+    dt : numpy.dtype
+        structure of the binary EEG+accelerometer+trigger data in the file
+    """
+    info = {}
+    # nedf files have some accelerometer channels sampled at 100Hz and
+    # several other channels sampled at 500Hz.
+    # The layout is
+    #  (100HzCh1S1, 100HzCh2S1, 100HzCh3S1),
+    # ((500HzCh1S1, 500HzCh2S1, …, 500HzChnS1),…,
+    #  (500HzCh1S2, 500HzCh2S2, …, 500HzChnS2), …
+    #  (500HzCh1S5,  500HzCh2S5, …, 500HzChnS5)),
+    #  (100HzCh1S2, 100HzCh2S2, 100HzCh3S2) and so on
+
+    # dtype for the binary data block
+    dt = []
+    # dtype for a single EEG sample
+    datadt = []
+    with open(filename, 'rb') as f:
+        header = f.read(10240)
+
+    headerend = header.find(b'\0')
+    if headerend == -1:
+        raise RuntimeError('End of header null not found')
+    headerxml = ElementTree.fromstring(header[:headerend])
+    nedfversion = headerxml.findtext('NEDFversion', '')
+    if nedfversion not in ['1.3', '1.4']:
+        print('Unexpected NEDFversion, hope this works anyway')
+    if headerxml.findtext('AdditionalChannelStatus', 'OFF') != 'OFF':
+        raise RuntimeError('Unknown additional channel, aborting.')
+
+    n_acc = int(headerxml.findtext('NumberOfChannelsOfAccelerometer', 0))
+    if n_acc:
+        # expect one sample of u16 accelerometer data per block
+        dt.append(('acc', '>u2', (n_acc,)))
+
+    eegset = headerxml.find('EEGSettings')
+    if eegset is None:
+        raise RuntimeError('No EEG channels found')
+    nchantotal = int(eegset.find('TotalNumberOfChannels').text)
+    info['nchan'] = nchantotal
+    info['sfreq'] = int(eegset.find('EEGSamplingRate').text)
+    info['ch_names'] = [e.text for e in eegset.find('EEGMontage')]
+    # expect nchantotal uint24s
+    datadt.append(('eeg', 'B', (nchantotal, 3)))
+
+    info['units'] = eegset.find('EEGUnits')
+
+    if headerxml.find('STIMSettings'):
+        # 2* -> two stim samples per eeg sample
+        datadt.append(('stim', 'B', (2, nchantotal, 3)))
+
+    if 'AdditionalChannelStatus' in headerxml:
+        raise RuntimeError('Unexpected AdditionalChannelStatus')
+
+    if headerxml.findtext('stepDetails/DeviceClass', '') == 'STARSTIM':
+        print('Found Starstim, not sure how to handle this')
```

Comment:
Again maybe use `warn`. Also, would it be better to abort in such a case? The message is not really reassuring and people might be worried that the result is not what they expect.

Reply:
From my experience with NE, NEDFVersion 1.3 and 1.4 and especially the stimulation data underwent some changes, but it mostly works. For some versions of the recording software and stimulation settings, the data block sizes are unexpected, but that's impossible to miss.